validator/src/state/merkle.rs

```diff
@@ -112,6 +115,59 @@ impl MerkleDatabase {
         })
     }
 
+    pub fn prune(db: &LmdbDatabase, merkle_root: &str) -> Result<bool, MerkleDatabaseError> {
```

Comment:
Why did you choose to make this an associated function instead of a method? Also, I am not a huge fan of returning a bool here. Can we return something more interesting like what was pruned (https://rust-lang-nursery.github.io/api-guidelines/flexibility.html#functions-expose-intermediate-results-to-avoid-duplicate-work-c-intermediate) or use a custom enum type in place of bool to convey meaning and promote flexibility: https://rust-lang-nursery.github.io/api-guidelines/type-safety.html#arguments-convey-meaning-through-types-not-bool-or-option-c-custom-type

Reply:
I made it an associated function, as it's not a an operation on an individual trie, but on the database as a whole.  It doesn't need to be a method of MerkleDatabase, since it doesn't need a state root, or a root node or any of that.

I can change the return type to the addresses of what was pruned, if that is more interesting.  Since it operates on one root at a time, returning was it pruned or not is all the information there is, beyond errors.